File: tests/fixtures/check_then_act/py/member_counter_fixture.py
# A counter checked before the await and raised after it bounds nothing. An unrelated member check
# in save() must not pair with a store write.
class Limiter:
    def __init__(self, limit, store):
        self.inflight = 0
        self.closed = False
        self.limit, self.store = limit, store

    async def run(self, task):
        if self.inflight >= self.limit:  # expect: check
            return
        await task()  # expect: gap await
        self.inflight += 1  # expect: write

    async def save(self, value):
        if self.closed:
            return
        await self.store.prepare(value)
        self.store.append(value)

File: tests/fixtures/check_then_act/ts/member_counter_fixture.ts
// A counter checked before the await and raised after it bounds nothing. An unrelated member check
// in save() must not pair with a store write.
export class Limiter {
  private inflight = 0;
  private closed = false;
  constructor(private readonly max: number, private readonly store: any) {}

  async run(task: () => Promise<void>) {
    if (this.inflight >= this.max) return; // expect: check
    await task(); // expect: gap await
    this.inflight++; // expect: write
  }

  async save(value: string) {
    if (this.closed) return;
    await this.store.prepare(value);
    this.store.append(value);
  }
}
